src/envoy/http/service_control/client_cache.cc

```diff
@@ -80,6 +81,14 @@ constexpr uint32_t kReportDefaultNumberOfRetries = 5;
 // The default value for network_fail_open flag.
 constexpr bool kDefaultNetworkFailOpen = true;
 
+// The service control call 4xx failure name, used in response code detail
+// later.
+constexpr char kScCallFail4XX[] = "4XX";
```

Comment:
4xx, lowercase. Then it will be consistent with our report labels

https://github.com/GoogleCloudPlatform/esp-v2/blob/fa28e9a6c337f82394f6fd4157c42a76417fbb6f/tests/utils/service_control_utils.go#L238

Reply:
Can we rename it to `http_call_4xx`?

These are used to say the actual HTTP call failed, not that Check returned a 4xx error status. I think just saying `4xx` is too general.

Reply:
On that note, why do we need to generalize to 4xx and 5xx? Why can't we just put the full status code?

`http_call_403`, `http_call_504`.

Reply:
Done. Use the status name instead of status code.